### PR TITLE
Update pre-hook-post-hook.md

### PR DESCRIPTION
**Pull request by mirnawong1**
clarify transactions not supported per [slack thread](https://dbt-labs.slack.com/archives/C05FWBP9X1U/p1695951000263009)

### CODE REVIEW

File: website/docs/reference/resource-configs/pre-hook-post-hook.md [hunk 1]
```diff
@@ -175,13 +175,16 @@ If multiple instances of any hooks are defined, dbt will run each hook using the
...
-* You want to insert a record into an audit <Term id="table" /> at the start of a run, and do not want that statement rolled back if the model creation fails.
+* You want to run a `VACUUM` in a `post-hook`, however, this cannot be executed within a transaction ([Redshift docs](https://docs.aws.amazon.com/redshift/latest/dg/r_VACUUM_command.html#r_VACUUM_usage_notes))
+* You want to insert a record into an audit <Term id="table" /> at the start of a run and do not want that statement rolled back if the model creation fails.
 
-To achieve this, you can use one of the following syntaxes. (Note: You should NOT use this syntax if using a database where dbt does not use transactions by default, including Snowflake, BigQuery, and Spark/Databricks.)
+To achieve this behavior, you can use one of the following syntaxes. (Important Note: Do not use this syntax if you are using a database where dbt does not use transactions by default. This includes databases like Snowflake, BigQuery, and Spark/Databricks.)
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/pre-hook-post-hook.md]:
```suggestion
To achieve this behavior, you can use one of the following syntaxes. (Important Note: Do not use this syntax if you are using a database where dbt does not use transactions by default. This includes databases like Snowflake, BigQuery, and Spark or Databricks.)
```

File: website/docs/reference/resource-configs/pre-hook-post-hook.md [hunk 2]
```diff
@@ -175,13 +175,16 @@ If multiple instances of any hooks are defined, dbt will run each hook using the
...
-* You want to insert a record into an audit <Term id="table" /> at the start of a run, and do not want that statement rolled back if the model creation fails.
+* You want to run a `VACUUM` in a `post-hook`, however, this cannot be executed within a transaction ([Redshift docs](https://docs.aws.amazon.com/redshift/latest/dg/r_VACUUM_command.html#r_VACUUM_usage_notes))
+* You want to insert a record into an audit <Term id="table" /> at the start of a run and do not want that statement rolled back if the model creation fails.
 
-To achieve this, you can use one of the following syntaxes. (Note: You should NOT use this syntax if using a database where dbt does not use transactions by default, including Snowflake, BigQuery, and Spark/Databricks.)
+To achieve this behavior, you can use one of the following syntaxes. (Important Note: Do not use this syntax if you are using a database where dbt does not use transactions by default. This includes databases like Snowflake, BigQuery, and Spark or Databricks.)
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/pre-hook-post-hook.md]:
```suggestion
To achieve this behavior, you can use one of the following syntaxes. 

Important note: Do not use this syntax if you are using a database where dbt does not use transactions by default. This includes databases like Snowflake, BigQuery, and Spark or Databricks.
```

File: website/docs/reference/resource-configs/pre-hook-post-hook.md [hunk 3]
```diff
@@ -175,13 +175,18 @@ If multiple instances of any hooks are defined, dbt will run each hook using the
...
+* You want to run a `VACUUM` in a `post-hook`, however, this cannot be executed within a transaction ([Redshift docs](https://docs.aws.amazon.com/redshift/latest/dg/r_VACUUM_command.html#r_VACUUM_usage_notes))
+* You want to insert a record into an audit <Term id="table" /> at the start of a run and do not want that statement rolled back if the model creation fails.
 
-To achieve this, you can use one of the following syntaxes. (Note: You should NOT use this syntax if using a database where dbt does not use transactions by default, including Snowflake, BigQuery, and Spark/Databricks.)
+To achieve this behavior, you can use one of the following syntaxes. 
+
+Important note: Do not use this syntax if you are using a database where dbt does not use transactions by default. This includes databases like Snowflake, BigQuery, and Spark or Databricks.
+
+<Tabs>
+<TabItem value="beforebegin" label="'before_begin' and 'after_commit'">
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/pre-hook-post-hook.md]:
```suggestion
<TabItem value="beforebegin" label="before_begin and 'after_commit">
```

File: website/docs/reference/resource-configs/pre-hook-post-hook.md [hunk 4]
```diff
@@ -200,6 +205,9 @@ select ...
 ```
 
 </File>
+</TabItem>
+
+<TabItem value="dictionary" label="Use a dictionary'">
```
Comment (mirnawong1) [inline website/docs/reference/resource-configs/pre-hook-post-hook.md]:
```suggestion
<TabItem value="dictionary" label="Use a dictionary">
```

**Comment by mirnawong1** [comment]
Waiting on team for feedback and will follow up this week

File: website/docs/reference/resource-configs/pre-hook-post-hook.md [hunk 5]
```diff
@@ -175,13 +175,17 @@ If multiple instances of any hooks are defined, dbt will run each hook using the
...
-* You want to insert a record into an audit <Term id="table" /> at the start of a run, and do not want that statement rolled back if the model creation fails.
+* You want to run a `VACUUM` in a `post-hook`, however, this cannot be executed within a transaction ([Redshift docs](https://docs.aws.amazon.com/redshift/latest/dg/r_VACUUM_command.html#r_VACUUM_usage_notes))
+* You want to insert a record into an audit <Term id="table" /> at the start of a run and do not want that statement rolled back if the model creation fails.
 
-To achieve this, you can use one of the following syntaxes. (Note: You should NOT use this syntax if using a database where dbt does not use transactions by default, including Snowflake, BigQuery, and Spark/Databricks.)
+To achieve this behavior, you can use one of the following syntaxes:
+  - Important note: Do not use this syntax if you are using a database where dbt does not use transactions by default. This includes databases like Snowflake, BigQuery, and Spark or Databricks.
```
Comment (patkearns10) [inline website/docs/reference/resource-configs/pre-hook-post-hook.md]:
```suggestion
  - Important note: Do not use this syntax if you are using a database where dbt does not support transactions. This includes databases like Snowflake, BigQuery, and Spark or Databricks.
```

**Review by patkearns10 — CHANGES_REQUESTED** [review]
Looks good to me, but I would suggest making this less open to interpretation as well.

**Comment by mirnawong1** [comment]
hey @patkearns10 , thanks so much for that and folded in your feedback - can you give it a look and approve? thank you again!

**Review by patkearns10 — APPROVED** [review]
🌵 LGTM!

**Pull request merged**